### PR TITLE
Add react component to story history, id and location clipboard

### PR DESCRIPTION
**Pull request by eddiecode**
Refactor Story's location and history `.input-form-group` into a React Component.

StoryHistoryLocation

### CODE REVIEW

File: app/assets/javascripts/components/story/StoryHistoryLocation.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+const StoryHistoryLocation = ({ id, link }) =>
+  <div className="col-xs-12 form-group input-group input-group-sm">
+    <input
+      id={`story-link-${id}`}
+      className="form-control input-sm"
+      value={link} readOnly="readonly"
```
Comment (marclerodrigues) [inline app/assets/javascripts/components/story/StoryHistoryLocation.js]: Why not breaking up this in different lines as you've done to the others props?

File: spec/features/stories_spec.rb [hunk 2]
```diff
@@ -47,6 +47,31 @@
...
+    before do
+      visit project_path(project)
+      wait_spinner
+      wait_page_load
+
+      find('.story-title').click
+      find('.toggle-history').click
+    end
+
+    it "turn history visible", js: true do
```
Comment (marclerodrigues) [inline spec/features/stories_spec.rb]: `it "turns history visible", js: true do`

**Comment by marclerodrigues** [comment]
LGTM :+1:

File: app/assets/javascripts/components/story/StoryHistoryLocation.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      value={link}
+    />
+
+    <span className="input-group-btn">
+      <button
+        className="btn btn-default btn-clipboard"
+        data-clipboard-target={`#story-link-${id}`}
+        type="button"
+      >
+        <img src="/clippy.svg" alt="Copy to clipboard" width="14px" />
```
Comment (talyssonoc) [inline app/assets/javascripts/components/story/StoryHistoryLocation.js]: Can we have a CSS for that width?

File: app/assets/javascripts/views/story_view.js [hunk 4]
```diff
@@ -365,44 +366,8 @@ module.exports = FormView.extend({
...
-        btn = this.make('button', {class: 'btn btn-default toggle-history'})
-        $(btn).html('<i class="mi md-18">history</i>');
-        $btnWrapper.append(btn);
-
-        $wrapper.append($btnWrapper[0]);
-        this.$el.append($wrapper[0]);
-
-        // activate the clipboard link
-        new Clipboard('.btn-clipboard');
+        const $storyHistoryLocation = $('<div data-story-history-location></div>');
```
Comment (talyssonoc) [inline app/assets/javascripts/views/story_view.js]: 👏  👏  👏

File: app/assets/javascripts/views/story_view.js [hunk 5]
```diff
@@ -567,6 +532,18 @@ module.exports = FormView.extend({
       this.$('[data-story-controls]').get(0)
     );
 
+    const historyLocationContainer = this.$('[data-story-history-location]').get(0);
+    if (historyLocationContainer) {
+      ReactDOM.render(
+        <StoryHistoryLocation
+          id={this.id}
+          link={`${this.getLocation()}#story-${this.id}`}
```
Comment (talyssonoc) [inline app/assets/javascripts/views/story_view.js]: Maybe `url` would be more descriptive than `link`, what do you think?

File: spec/features/stories_spec.rb [hunk 6]
```diff
@@ -47,6 +47,31 @@
 
   end
 
+  describe "story history" do
+
+    let!(:story) do
```
Comment (talyssonoc) [inline spec/features/stories_spec.rb]: Since you're not using the `story` variable, you could put it inside a `before`.

**Pull request merged**